commit 11068f99bb8d623f9590c406a49722d0b3d2bc63 (refs/changes/24/4724/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-08-20T14:08:36-07:00 (1 year, 2 months ago)
    
    TRITON-707 CNS crash on access to query.src because redis claim() took too long, remote closed TCP connection

diff --git a/lib/dns-server.js b/lib/dns-server.js
index c4c323b..ce099cf 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -236,6 +236,12 @@ DNSServer.prototype.handleQuery = function (q, cb) {
 
 	q.log.trace('begin query');
 
+	if (q.src === undefined || q.src === null) {
+		q.log.warn('dropping query from already-closed connection');
+		cb();
+		return;
+	}
+
 	var name = q.name().toLowerCase();
 	var type = q.type();
 	var addr = q.src.address;
